psi4/driver/driver.py

```diff
@@ -873,6 +873,158 @@ def properties(*args, **kwargs):
+            raise ValueError("No scan!")
+        Cons, CVals = geometric.optimize.ParseConstraints(M, constraints_string)
+    
+    # Set up the internal coordinate system
+    coordsys = geometric_opts.get('coordsys', 'tric')
+    CoordSysDict = {
+        'cart': (geometric.internal.CartesianCoordinates, False, False),
+        'prim': (geometric.internal.PrimitiveInternalCoordinates, True, False),
+        'dlc': (geometric.internal.DelocalizedInternalCoordinates, True, False),
+        'hdlc': (geometric.internal.DelocalizedInternalCoordinates, False, True),
+        'tric': (geometric.internal.DelocalizedInternalCoordinates, False, False)
+    }
+    
+    # Build internal coordinates
+    CoordClass, connect, addcart = CoordSysDict[coordsys.lower()]
+    IC = CoordClass(
+        M,
+        build=True,
+        connect=connect,
+        addcart=addcart,
+        constraints=Cons,
+        cvals=CVals[0] if CVals is not None else None)
+    
+    # Get initial coordinates in bohr
+    coords = M.xyzs[0].flatten() / qcel.constants.bohr2angstroms
+
+    # Setup an optimizer object
+    params = geometric.optimize.OptParams(**geometric_opts)
+    optimizer = geometric.optimize.Optimizer(coords, M, IC, engine, None, params)
+    
+    # TODO: print constraints
+    # IC.printConstraints(coords, thre=-1)
+    optimizer.calcEnergyForce()
+    optimizer.prepareFirstStep()
+    grms, gmax = optimizer.calcGradNorm()
+    conv_gmax = '*' if gmax < params.Convergence_gmax else ' '
+    conv_grms = '*' if grms < params.Convergence_grms else ' '
+    core.print_out('\n')
+    core.print_out("\n  ==> geomeTRIC Optimizer <==\n")
+    core.print_out("\n  Measures of convergence in internal coordinates in au.")
+    core.print_out("\n  Criteria marked as inactive (o), active & met (*), and active & unmet ( ).")
+    core.print_out("\n  ---------------------------------------------------------------------------------------------")
+    core.print_out("\n   Step     Total Energy     Delta E     MAX Force     RMS Force      MAX Disp      RMS Disp   ")
+    core.print_out("\n  ---------------------------------------------------------------------------------------------")
+    core.print_out((f"\n    Convergence Criteria  {params.Convergence_energy:10.2e}    "
+                    f"{params.Convergence_gmax:10.2e}    {params.Convergence_grms:10.2e}    "
+                    f"{params.Convergence_dmax:10.2e}    {params.Convergence_drms:10.2e}"))
+    core.print_out("\n  ---------------------------------------------------------------------------------------------")
+
+    core.print_out((f"\n   {optimizer.Iteration:4d} {optimizer.E:16.8e}    -------- *  "
+                    f"{gmax:10.2e} {conv_gmax}  {grms:10.2e} {conv_grms}    -------- *    -------- *"))
+    while True:
+        if optimizer.state in [geometric.optimize.OPT_STATE.CONVERGED, geometric.optimize.OPT_STATE.FAILED]:
+            core.print_out("\n\n  Optmization converged!\n")
```

Comment:
Wrong spelling. Should we be printing "Optimization converged" in case the `OPT_STATE.FAILED` case?